pkg/core/handler/api.go

```diff
@@ -13,39 +14,42 @@ type APIHandler interface {
 }
 
 type APIResponse struct {
-	Result interface{} `json:"result,omitempty"`
-	Err    error       `json:"error,omitempty"`
+	Result interface{}  `json:"result,omitempty"`
+	Err    skyerr.Error `json:"error,omitempty"`
 }
 
 func APIHandlerToHandler(apiHandler APIHandler) Handler {
 	return HandlerFunc(func(rw http.ResponseWriter, r *http.Request, ctx AuthContext) {
+		httpStatus := http.StatusOK
+		response := APIResponse{}
+		encoder := json.NewEncoder(rw)
+
+		defer func() {
+			if response.Err != nil {
+				httpStatus = nextSkyerr.ErrorDefaultStatusCode(response.Err)
+			}
+
+			rw.Header().Set("Content-Type", "application/json")
+			rw.WriteHeader(httpStatus)
+			encoder.Encode(response)
```

Comment:
To prevent writing the response multiple times, should avoid writing response in defer function. We did it in `RecoverMiddleware`.